internal/sgbouncewizard/sgbouncewizard.go

```diff
@@ -242,41 +255,137 @@ func (srv *Server) updateRuleRoute(w http.ResponseWriter, r *http.Request) {
 
 	err = srv.DBClient.UpdateRule(&newRule)
 
+	if err != nil {
+		log.Println(err)
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
+
+	err = srv.DBClient.CreateChangeLogEntry(newRule.ID, &newRule)
+
 	if err != nil {
 		log.Println(err)
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
 
-	w.Header().Set("Access-Control-Allow-Origin", "*")
 	w.Header().Set("Content-Type", "application/json")
 	w.WriteHeader(http.StatusOK)
 }
 
+func (srv *Server) ChangelogContext(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var changelog []models.BounceRule
+
+		bounce_id := chi.URLParam(r, "bounce_id")
+
+		bouncd_idInt, err := strconv.Atoi(bounce_id)
+
+		if err != nil {
+			log.Println(err)
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		queryParams := r.URL.Query()
+
+		limit, ok := queryParams["limit"]
 
-func (srv *Server) GetChangelog(w http.ResponseWriter, r *http.Request) {
-	rules, err := srv.DBClient.GetAllChangelogEntries()
+		if !ok {
+			changelog, err = srv.DBClient.GetChangeLogEntries(bouncd_idInt, nil)
+		} else {
+
+			limitAsString := strings.Join(limit, "")
```

Comment:
What is going on here? I think `limit` is a slice of strings? If so perhaps we reject the request if more than one limit is submitted

Reply:
The limit is suppose to be a number. Later on i convert from a string to int and if they did not put an int as the value to limit it will be caught in the error.

Reply:
https://github.com/jimmyjames85/bouncecm/blob/7adcea3ec5031cf2baf02016445c268478015d2b/internal/sgbouncewizard/sgbouncewizard.go#L292

`limit` is a `[]string` and you are joining the slice into one string with `limitAsString := strings.Join(limit, "")`. I suggest rejecting the request unless the length of `limit` is exactly one.

Reply:
Consider using `limit, err := strconv.Atoi(r.FormValue("limit"))`

https://golang.org/src/net/http/request.go?s=40143:40189#L1260